### ISSUE TITLE
[preset-env] Android not supporting anything?

### ISSUE THREAD

**Issue by luzat** [babel/babel#8351]
## Bug Report
The preset seems to think that Android does not support anything and enables every plugin because of `"android": "67"` support. Chrome for Android should support most features in that version.

I set up an [example repository](https://github.com/luzat/babel-preset-env-demo) with basic configuration and debug enabled. Output after `npm install`/`yarn install` and `npm run demo`/`yarn run demo` is:

```
> babel-preset-env-demo@1.0.0 demo /home/tom/p/babel-preset-env-demo
> babel index.js

@babel/preset-env: `DEBUG` option

Using targets:
{
  "android": "67",
  "chrome": "66",
  "edge": "17",
  "firefox": "60",
  "ios": "11",
  "safari": "11.1"
}

Using modules transform: commonjs

Using plugins:
  transform-template-literals { "android":"67" }
  transform-literals { "android":"67" }
  transform-function-name { "android":"67", "edge":"17" }
  transform-arrow-functions { "android":"67" }
  transform-block-scoped-functions { "android":"67" }
  transform-classes { "android":"67" }
  transform-object-super { "android":"67" }
  transform-shorthand-properties { "android":"67" }
  transform-duplicate-keys { "android":"67" }
  transform-computed-properties { "android":"67" }
  transform-for-of { "android":"67" }
  transform-sticky-regex { "android":"67" }
  transform-dotall-regex { "android":"67", "edge":"17", "firefox":"60", "ios":"11" }
  transform-unicode-regex { "android":"67" }
  transform-spread { "android":"67" }
  transform-parameters { "android":"67" }
  transform-destructuring { "android":"67", "edge":"17" }
  transform-block-scoping { "android":"67" }
  transform-typeof-symbol { "android":"67" }
  transform-new-target { "android":"67" }
  transform-regenerator { "android":"67" }
  transform-exponentiation-operator { "android":"67" }
  transform-async-to-generator { "android":"67" }
  proposal-async-generator-functions { "android":"67", "edge":"17", "ios":"11", "safari":"11.1" }
  proposal-object-rest-spread { "android":"67", "edge":"17", "ios":"11" }
  proposal-unicode-property-regex { "android":"67", "edge":"17", "firefox":"60", "ios":"11" }

Using polyfills: No polyfills were added, since the `useBuiltIns` option was not set.
"use strict";
```

I would expect the Android browser to support most if not all of these. I guess there is a mapping from Android to the equivalent Chrome versions missing at some point (compat-table, caniuse, browserslist?) and I recall having used Babel 7 previously without this behavior. I imagine some third-party library update to be responsible.

**babel.config.js**
```js
module.exports = {
  presets: [
    [
      '@babel/env',
      {
        debug: true,
      },
    ],
  ],
}
```

I am not sure if the feature sets of Android x and Chrome x are equivalent starting with some version of if there are numbering and feature differences to account for, but probably either compat-table or preset-env should have some mapping., most likely for all versions of Android >4.4.

**Environment**
- Babel version(s): 7.0.0-beta.54; downgrading to beta.31 did not help.
- Node/npm version: Node v10.7.0
- OS: Debian GNU/Linux
- How you are using Babel: cli and loader

**Comment by existentialism** [comment]
> I recall having used Babel 7 previously without this behavior

Hmm, not sure as compat-table only has explicit handling of android <= 4.4.3, and removed the Chrome aliases for a while.

Ref:
https://github.com/kangax/compat-table/pull/1104

My initial reaction is I'd rather keep it (relatively) simple and consistent with our data source (compat-table), and ask users to use `Chrome`, than maintain a mapping.

**Comment by luzat** [comment]
It is very non-intuitive if specifying Android as a platform includes all plugins/transforms without explanation. Also, other tools might require Android to be in the browser list, which would lead to maintaining multiple lists for different tools.

There is already special code for [Opera and Electron](https://github.com/babel/babel/blob/adae1501aca6b6107477ea61b47f72cee43a4438/packages/babel-preset-env/scripts/build-data.js#L259]), too, which could be removed with the same reasoning.

Looking at the [version history](https://en.wikipedia.org/wiki/Google_Chrome_version_history) it seems that after 4.4, starting with Chrome 18, the same engine has always been used. The code next to Opera's and Electron's code can probably be easily extended:

```js
// add android
if (plugin.chrome >= 18) {
  plugin.android = plugin.chrome;
}
```

Untested; but it looks as if this could suffice?

**Comment by existentialism** [comment]
I guess I view `Android 57` a bit differently than `Opera 57` or `Electron 57`... since Android 57 doesn't exist? It's Chrome for Android v57... while it might be pedantic; since we haven't had any complaints about this, I'm a bit hesitant to introduce this change.

**Comment by luzat** [comment]
I need to think a bit about this. caniuse knows about Android browsers <=4.4 and Android 67 (with global usage of 0), which I think are encoded as "android". Chrome for Android should be encoded as `and_chr`, currently at 67, too. For some reason browserslist does not return `and_chr 67`, but `android 67`. This could be the root cause. `and_chr` would be mapped to chrome by preset-env.

This may be a bug in some of the other components (browserslist?) which my repository above can be used for to reproduce. Basically, a query to browserslist with `>1%` should include `and_chr 67`, not `android 67`. I will try to look into this, but if somebody could confirm that android is falsely returned, that would be helpful.

**Comment by jhuesos** [comment]
I just faced this issue as well. If I add `last 2 Android major versions` to my target, `android 67` is added to the target. And if I do that, everything gets compiled to ES5, even though according to https://caniuse.com/#feat=es6-class, Android 67 supports ES6 Classes.

**Comment by stefanprobst** [comment]
This is especially problematic because the browserslist defaults include Android 67:
```bash
$ npx browserslist defaults | grep -i android
android 67
android 4.4.3-4.4.4
```

**Comment by melfa** [comment]
Any updates on this? browserslist defaults still returns android 76, preset-env with debug enabled reports "android": "4" and compiles all to ES5.

```
% npx browserslist defaults
and_chr 79
and_ff 68
and_qq 1.2
and_uc 12.12
android 76
baidu 7.12
chrome 79
chrome 78
chrome 49
edge 18
edge 17
firefox 72
firefox 71
firefox 70
firefox 68
ie 11
ios_saf 13.3
ios_saf 13.2
ios_saf 13.0-13.1
ios_saf 12.2-12.4
kaios 2.5
op_mini all
op_mob 46
opera 64
opera 63
safari 13
safari 12.1
safari 5.1
samsung 10.1
samsung 9.2
```

```
@babel/preset-env: `DEBUG` option

Using targets:
{
  "android": "80",
  "chrome": "4",
  "edge": "18",
  "firefox": "71",
  "ios": "11.3",
  "opera": "65",
  "safari": "13"
}
```

**Comment by JLHwung** [comment]
This issue has been fixed in #10814, please update `@babel/preset-env` to 7.8.4 or above.
